src/plugins/ui_actions/public/actions/action_internal.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Action, ActionExecutionContext, AnyActionDefinition } from './action';
+import { Presentable } from '../util/presentable';
+import { createActionStateContainer, ActionState } from './action_state_container';
+import { uiToReactComponent } from '../../../kibana_react/public';
+import { ActionContract } from './action_contract';
+
+export class ActionInternal<A extends AnyActionDefinition>
+  implements Action<ActionExecutionContext<A>>, Presentable<ActionExecutionContext<A>> {
```

Comment:
You are right, it should not need to implement `Action` but I put it here on purpose for backwards compatibility, because all users of `ui_actions` plugin expect to register items of type `Action` and receive back items of type `Action`, too. By making `ActionDefinition` and `ActionInternal` implement `Action` I'm basically ensuring that those still support everything that `Action` used to support.

But going forwards, as `Action` interface will be removed we will also be able to remove this `implements Action`. At least that was my idea.